[DM-1101] New All devices view (#523)
SoftwareAG/c8y-docs: content/users-guide/device-management-bundle/viewing-devices.md

</tbody>
</table>

> The devices list displays up to 100 rows. If a list contains more than 100 devices, click **Load more** at the bottom of the list to display the next 100 entries.

Reviewer: Do we want to mention here that list is paginated and only for users with restricted permissions Load more button will be available?

Author: Added a paragraph here https://github.com/SoftwareAG/c8y-docs/pull/523/commits/9f4f99b766a43d09f4450a23872b006022ab6c34

### Configuring columns

The columns shown in the device list may be configured to your needs.

#### To show/hide standard columns

1. In the table header, click **Configure columns**.
2. In the resulting dropdown, select/clear the checkboxes for all columns as required. <br>![Configure columns](/images/users-guide/DeviceManagement/devmgmt-device-list-configure-columns.png)
<br>

The device list will reflect your changes and only show the selected columns.

#### To add custom columns

Moreover, you can add custom columns to show additional device properties.

1. In the **Configure columns** dropdown, click **Add custom column**.<br>
![Configure columns](/images/users-guide/DeviceManagement/devmgmt-device-list-custom-column.png)<br>
2. In the **Header** field, enter a header for the new custom column.  
3. In the **Fragment path** field, enter the property of the device to be shown. Nested properties will be accepted. However, for nested properties its only possible to select {{< product-c8y-iot >}} standard fragments like `c8y_Mobile.mcc`.

> 4. Toggle the **Add another column after saving this one** button active to create another custom column right after saving the current one without leaving the dialog.

Reviewer 2: ```suggestion
4. Switch the **Add another column after saving this one** toggle to active to create another custom column right after saving the current one without leaving the dialog.
```

</tbody>
</table>

> For users with global roles the devices list will display the items in a paginated manner where you are able to select the number of items per page and directly jump to any page. Users with inventory roles will see up to 50 items initially. If there are more than 50 devices available to you more will be loaded as you scroll down the list.

Reviewer 2: ```suggestion
For users with global roles the devices list displays the items in a paginated manner. You can select the number of items per page and directly jump to any page. Users with inventory roles see up to 50 items initially. If there are more than 50 devices available more devices are loaded as you scroll down the list.
```

![Device filtering](/images/users-guide/DeviceManagement/devmgmt-devices-filter.png)

> In the resulting popup, specify your filter options.

Reviewer 2: ```suggestion
In the resulting dialog box, specify your filter options.
```

Reviewer 2: Ok, then let´s change the sentence to "Specify your filter options in the dropdown list." We usually do not use the term popup as it is not very precise.

Most columns represent text fields. You can filter these columns by simply entering an arbitrary text into the textbox as in the search field. Click **+ Add next** to add another textbox if you want to filter for more than one term.

Apart from filtering for text there are several other options:

* In case of date fields (for example **Registration date**), you specify a date range to filter for.
* In the **Status** column you can filter for various criteria representing the send, push or maintenance status of the device.
* In the **Alarm** column the filtering options you may select correspond to the alarm types (critical, major, minor, warning, no alarms).

> * For custom columns, you can choose between filtering based on whether the fragment exists by clicking the **Only rows where value is defined** button active or specify one or more filter terms the entry must match.

Reviewer 2: ```suggestion
* For custom columns, select **Only rows where value is defined** to filter based on whether the fragment exists,  or specify one or more filter terms the entry must match.
```

Finally, click **Apply** to carry out the filtering.

The devices list will now only display devices matching the filtering options.

> Sorting is available on every column. Click the sort icon next in column header once to sort entries in ascending order and twice to sort the entries in descending order. One more click on the sort icon will remove sorting for this column.

Reviewer 2: ```suggestion
Sorting is available on every column. Click the sort icon in the respective column header once to sort the entries in ascending order and twice to sort the entries in descending order. Click the sort icon once more to remove sorting for this column.
```

> Click **Clear filters** in table header if you want to clear all filters and view all devices.

Reviewer 2: ```suggestion
Click **Clear filters** in the table header if you want to clear all filters and view all devices.
```

Reviewer 2: The devmgmt-devices-connectionstatus image does not have the right browser size. The full screen should not be larger than 1440 pixel, but this image does not show the full screen.

Author: Replaced with a new screenshot.
